src/android/CodePush.java

```diff
@@ -461,6 +487,8 @@ public void onStart() {
                 handleAppStart();
                 this.markUpdate();
                 this.codePushPackageManager.clearPendingInstall();
+            } else if (codePushReportingManager.hasFailedReport()) {
```

Comment:
why can we only report a failed update on resume? why not on restart? are we waiting for the installation to be confirmed?